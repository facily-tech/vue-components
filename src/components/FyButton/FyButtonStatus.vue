<template>
  <fy-button
    :color="color"
    @click="$emit('active-control', { key: event, value: item })"
    data-test="btn-status"
    x-small
    text
    v-bind="propsAndAttrs"
    v-on="$listeners"
  >
    <span class="text-uppercase">{{ name }}</span>
    <v-icon small class="pl-1">mdi-circle</v-icon>
  </fy-button>
</template>

<script lang="ts">
import Vue from 'vue';
import FyButton from './FyButton.vue';

export default Vue.extend({
  name: 'FyButtonStatus',

  components: {
    FyButton,
  },

  props: {
    item: {
      type: Object,
    },

    status: {
      type: Boolean,
      default: true,
    },
  },

  computed: {
    propsAndAttrs(): Record<string, unknown> {
      return {
        ...this.$attrs,
        ...this.$props,
      };
    },

    name() {
      return this.status ? 'Ativo' : 'Inativo';
    },

    color() {
      return this.status ? 'success' : 'warning';
    },

    event() {
      return !this.status ? 'actitve' : 'inactitve';
    },
  },
});
</script>
