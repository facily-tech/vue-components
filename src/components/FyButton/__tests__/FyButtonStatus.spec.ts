import Vue from 'vue';
import Router from 'vue-router';
import FyButtonStatus from '../FyButtonStatus.vue';
import { createLocalVue, mount, Wrapper } from '@vue/test-utils';
import Vuetify from 'vuetify';

Vue.use(Vuetify);

describe('FyButtonSearch', () => {
  // eslint-disable-next-line
  let router: Router;
  let localVue: typeof Vue;
  const event = 'active-control';

  beforeEach(() => {
    router = new Router();
    localVue = createLocalVue();
    localVue.use(Router);
  });

  it('should render component and match snapshot', () => {
    const wrapper = mount(FyButtonStatus, {
      localVue,
      vuetify: new Vuetify(),
      router,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('when status is active', () => {
    let wrapper: Wrapper<any>;

    beforeEach(() => {
      wrapper = mount(FyButtonStatus, {
        localVue,
        vuetify: new Vuetify(),
        router,
        propsData: {
          item: { id: 1 },
          status: true,
        },
      });
    });

    it('name', () => {
      expect(wrapper.vm.name).toBe('Ativo');
    });

    it('color', () => {
      expect(wrapper.vm.color).toBe('success');
    });

    it('event', () => {
      expect(wrapper.vm.event).toBe('inactitve');
    });

    it('when button is clicked', () => {
      expect(wrapper.emitted('click-status')).toBeFalsy();

      wrapper.find('[data-test="btn-status"]').trigger('click');

      expect(wrapper.emitted(event)).toBeTruthy();
      expect(wrapper.emitted(event)[0]).toEqual([
        {
          key: 'inactitve',
          value: {
            id: 1,
          },
        },
      ]);
    });
  });

  describe('when status is inactitve', () => {
    let wrapper: Wrapper<any>;

    beforeEach(() => {
      wrapper = mount(FyButtonStatus, {
        localVue,
        vuetify: new Vuetify(),
        router,
        propsData: {
          item: { id: 1 },
          status: false,
        },
      });
    });

    it('name', () => {
      expect(wrapper.vm.name).toBe('Inativo');
    });

    it('color', () => {
      expect(wrapper.vm.color).toBe('warning');
    });

    it('event', () => {
      expect(wrapper.vm.event).toBe('actitve');
    });

    it('when button is clicked', () => {
      expect(wrapper.emitted('click-status')).toBeFalsy();

      wrapper.find('[data-test="btn-status"]').trigger('click');

      expect(wrapper.emitted(event)).toBeTruthy();
      expect(wrapper.emitted(event)[0]).toEqual([
        {
          key: 'actitve',
          value: {
            id: 1,
          },
        },
      ]);
    });
  });
});
