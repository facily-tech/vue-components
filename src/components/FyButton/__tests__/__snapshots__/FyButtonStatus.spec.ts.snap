// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyButtonSearch should render component and match snapshot 1`] = `<button type="button" class="v-btn v-btn--text theme--light elevation-0 v-size--x-small success--text" style="min-height: 40px;" data-test="btn-status" status="true" color="success" x-small="" text=""><span class="v-btn__content"><span class="text-uppercase">Ativo</span> <i aria-hidden="true" class="v-icon notranslate pl-1 mdi mdi-circle theme--light" style="font-size: 16px;"></i></span></button>`;
